.github/workflows/api-check.yml

```diff
@@ -20,12 +18,35 @@ jobs:
         with:
           fetch-depth: 0
 
-      # Install nightly (stable is already installed)
+      # Install nightly
+      # The nightly version needs to match the cargo-public-api version.
+      # Check here: https://github.com/Enselic/cargo-public-api#compatibility-matrix
       - uses: actions-rs/toolchain@v1
         with:
-          toolchain: nightly
+          toolchain: nightly-2022-09-06
           profile: minimal
 
-      # Install and run cargo public-api and deny any API diff
-      - run: cargo install cargo-public-api
-      - run: cargo public-api --diff-git-checkouts ${GITHUB_BASE_REF} ${GITHUB_HEAD_REF} --deny=all
+      # Install a specific version of cargo-public-api
+      - name: Install cargo-public-api
+        run: cargo install cargo-public-api --version 0.17.0
+      # We cannot compare two branches using 'cargo public-api --diff-git-checkouts ${GITHUB_BASE_REF} ${GITHUB_HEAD_REF}',
+      # as the Github checkout action does not actually fetch the branch from the forked repo, and cargo-public-api cannot
+      # check out ${GITHUB_HEAD_REF} if the PR is from a fork.
+      # So we get API dumped to text files, and diff the text files.
+      - name: Dump public API for PR
+        run: |
+          cargo +nightly-2022-09-06 public-api > pr-api.txt
```

Comment:
If we are sure the current HEAD is the PR commit, can we simply do the following?
```
cargo public-api --diff-git-checkouts ${GITHUB_BASE_REF} HEAD --deny=all
```

Reply:
We figured out that `${{ github.event.pull_request.head.sha }}` works fine here.